packages/driver/cypress/e2e/commands/actions/type.cy.js

```diff
@@ -3205,4 +3157,312 @@ describe('src/cy/commands/actions/type - #type', () => {
+
+          cy.get('@spyChange').should('have.been.calledOnce')
+          cy.get('@spyInput').should('have.been.calledOnce')
+        })
+      }
+    })
+
+    context('input[type=datetime-local]', () => {
+      if (!isWebKit) {
+        it('{upArrow} increases time value', () => {
+          cy.get('#datetime-local-without-value').type('{upArrow}')
+        })
+
+        it('{downArrow} decreases time value', () => {
+          cy.get('#datetime-local-without-value').type('{downArrow}')
+        })
+
+        it('{upArrow} triggers events on input', () => {
+          cy.get('#datetime-local-with-value')
+          .then(($input) => {
+            $input.on('change', cy.spy().as('spyChange'))
+            $input.on('input', cy.spy().as('spyInput'))
+
+            return $input
+          })
+          .type('{upArrow}')
+
+          cy.get('@spyInput').should('have.been.calledOnce')
+          cy.get('@spyChange').should('have.been.calledOnce')
+        })
+
+        it('{downArrow} triggers events on input', () => {
+          cy.get('#datetime-local-with-value')
+          .then(($input) => {
+            $input.on('change', cy.spy().as('spyChange'))
+            $input.on('input', cy.spy().as('spyInput'))
+
+            return $input
+          })
+          .type('{downArrow}')
+
+          cy.get('@spyChange').should('have.been.calledOnce')
+          cy.get('@spyInput').should('have.been.calledOnce')
+        })
+      }
+    })
+
+    context('input[type=number]', () => {
+      it('{upArrow} increases number value', () => {
+        cy.get('#number-without-value').then(($input) => $input.val(0))
+        cy.get('#number-without-value').type('{upArrow}')
+        cy.get('#number-without-value').should('have.value', 1)
+      })
+
+      it('{downArrow} decreases number value', () => {
+        cy.get('#number-without-value').then(($input) => $input.val(1))
+        cy.get('#number-without-value').type('{downArrow}')
+        cy.get('#number-without-value').should('have.value', 0)
+      })
+
+      it('{upArrow} triggers events on input', () => {
+        cy.get('#number-with-value')
+        .then(($input) => {
+          $input.on('change', cy.spy().as('spyChange'))
+          $input.on('input', cy.spy().as('spyInput'))
+
+          return $input
+        })
+        .type('{upArrow}')
+
+        cy.get('@spyInput').should('have.been.calledOnce')
+        cy.get('@spyChange').should('have.been.calledOnce')
+      })
+
+      it('{downArrow} triggers events on input', () => {
+        cy.get('#number-with-value')
+        .then(($input) => {
+          $input.on('change', cy.spy().as('spyChange'))
+          $input.on('input', cy.spy().as('spyInput'))
+
+          return $input
+        })
+        .type('{downArrow}')
+
+        cy.get('@spyChange').should('have.been.calledOnce')
+        cy.get('@spyInput').should('have.been.calledOnce')
+      })
+    })
+
+    context('input[type=range]', () => {
+      if (!isWebKit) {
```

Comment:
I experienced certain issues with testing input[type=range] with Webkit browser. 

Issue was with with following test case:

`
cy.get('#range-without-value').then(($input) => $input.val(0)) /* Apparently this part of the code was failing to actually set value to one be */

cy.get('#range-without-value').type('{upArrow}') /* This part is supposed to be increasing value by 1, hence value should be 1'

cy.get('#range-without-value').should('have.value', 1)
`
This code perfectly works for other browsers, but what actually happens with Webkit browser is that instead of getting value of 1, I am actually getting value of 51 (Webkit based browsers have default value for range inputs of 50, so apparently value is not getting set to 0 inititally)